docs/testing/unit-testing.md

```diff
@@ -0,0 +1,406 @@
+1. Require (or import) the library to your test file. The following code is an example:
+
+   ```javascript
+   const OfficeMockObject = require("office-addin-mock");
+   ```
+
+1. Require (or import) the module that contains the add-in function that you want to test. The following is an example that assumes your test file is in a subfolder the folder with your add-in's code files:
+
+   ```javascript
+   const myOfficeAddinFeature = require("../my-office-add-in");
+   ```
+
+1. Create a data object that has the properties and subproperties that you need to mock to test the function. The following is an example of an object that mocks the Excel [Workbook.range.address](/javascript/api/excel/excel.range#address) property and the [Workbook.getSelectedRange](/javascript/api/excel/excel.workbook#getSelectedRange__) method. This isn't the final mock object. Think of it as a seed object that is used by `OfficeMockObject` to create the final mock object.
+
+   ```javascript
+   const mockData = {
+     workbook: {
+       range: {
+         address: "C2:G3",
+       },
+       getSelectedRange: function () {
+         return this.range;
+       },
+     },
+   };
+   ```
+
+1. Pass the data object to the `OfficeMockObject` constructor. Note the following about the returned `OfficeMockObject` object:
+
+   - It is a simplified mock of an [OfficeExtension.ClientRequestContext](/javascript/api/office/officeextension.clientrequestcontext) object. (It is also possible to mock a host object, such as [Excel](/javascript/api/excel) or [Word](/javascript/api/word). For an example, see [Mocking a host object](#mocking-a-host-object).)
+   - The mock object has all the members of the data object and also has mock `load()` and `sync()` methods.
+   - The mock object will mimic crucial error behavior of the `ClientRequestContext` object. For example, if the Office API you are testing tries to read a property without first loading the property and calling `sync()`, then the test will fail with an error similar to what would be thrown in production runtime: "Error, property not loaded".
+
+   ```javascript
+   const contextMock = new OfficeMockObject.OfficeMockObject(mockData);
+   ```
+
+    > [!NOTE]
+    > Full reference documentation for the `OfficeMockObject` type is at [Office-Addin-Mock](https://github.com/OfficeDev/Office-Addin-Scripts/tree/master/packages/office-addin-mock).
+
+1. In the syntax of your test framework, add a test of the function, but use the `OfficeMockObject` object in place of the object that it mocks, in this case the `ClientRequestContext` object. The following continues the example in Jest. This example test assumes that the add-in function that is being tested is called `getSelectedRangeAddress`, that it takes a `ClientRequestContext` object as a parameter, and that it is intended to return the address of the currently selected range. The full example is at [Mocking a ClientRequestContext object](#mocking-a-clientrequestcontext-object).
```

Comment:
Small suggestion to help people keep context. This might go against style guidelines, so grain of salt and all that.
```suggestion
1. In the syntax of your test framework, add a test of the function, but use the `OfficeMockObject` object in place of the object that it mocks, in this case the `ClientRequestContext` object. The following continues the example in Jest. This example test assumes that the add-in function that is being tested is called `getSelectedRangeAddress`, that it takes a `ClientRequestContext` object as a parameter, and that it is intended to return the address of the currently selected range. The full example is [later in this article](#mocking-a-clientrequestcontext-object).
```